src/main/java/seedu/address/model/person/Vendor.java

```diff
@@ -0,0 +1,56 @@
+package seedu.address.model.person;
+
+import java.util.Set;
+
+import seedu.address.model.tag.Tag;
+import seedu.address.model.wedding.Wedding;
+
+/**
+ * Represents a Vendor in the WedLinker.
+ * Guarantees: details are present and not null, field values are validated, immutable.
+ */
+public class Vendor extends Person {
+
+    /**
+     * Every field must be present and not null.
+     */
+    // add the Task field here
+    public Vendor(Name name, Phone phone, Email email, Address address, Set<Tag> tags, Set<Wedding> weddings) {
+        super(name, phone, email, address, tags, weddings);
+    }
```

Comment:
The fields are similar to Person and can be optional right?

Reply:
Yea they can be optional.